Bio/SeqUtils/__init__.py

```diff
@@ -667,6 +668,40 @@ def calculate(self, sequence):
 
         return exp(cai_value / cai_length)
 
+    def optimize(self, sequence, seq_type="DNA"):
+        """Return a new DNA sequence with preferred codons only."""
+        try:  # If seq record is provided, convert to sequence
+            sequence = sequence.seq
+        except AttributeError:  # not a  SeqRecord object
+            pass
+        seq = sequence.upper()
+        # Make dict with amino acids referencing preferred codons
+        pref_codons = {}
+        for codon, aminoacid in self._table.forward_table.items():
+            if self[codon] == 1.0:
+                if aminoacid in pref_codons:
+                    message = f"{pref_codons[aminoacid]} and {codon} are equally preferred. Using {codon}"
+                    warnings.warn(message, RuntimeWarning)
```

Comment:
Note that by default, the warning will be shown only once in a single Python session for a specific aminoacid and codon. Calling optimize twice with the same sequence would show the warning only the first time. Also, if a subsequent sequence contains the same codon, then again the warning will not be shown.
A better way is to have strict=True|False argument that specifies if an Exception should be raised if equally preferred codons are found.

Reply:
I wouldn't object to the suggested strict mode, but personally am content with the proposed warning behaviour.